test/shork_test.gleam

```diff
@@ -5,103 +5,371 @@ import gleeunit/should
+  let assert Ok(returned) =
+    shork.query(sql)
     |> shork.parameter(shork.int(1))
     |> shork.returning({
-      use x0 <- decode.field(0, decode.int)
-      use x1 <- decode.field(1, decode.string)
-      use x2 <- decode.field(2, decode.int)
+      use id <- decode.field(0, decode.int)
+      use name <- decode.field(1, decode.string)
+      use size <- decode.field(2, decode.float)
 
-      decode.success(#(x0, x1, x2))
+      decode.success(#(id, name, size))
     })
     |> shork.execute(connection)
 
-  should.equal(a.rows, [#(1, "Tony Shark", 54)])
+  should.equal(returned.column_names, ["id", "name", "size"])
+  list.length(returned.rows) |> should.equal(1)
+
+  returned.rows
+  |> should.equal([#(1, "Hainelore", 90.0)])
 }
 
-pub fn no_table_test() {
-  let connection =
-    shork.default_config()
-    |> shork.user("root")
-    |> shork.password("strong_password")
-    |> shork.database("poke")
-    |> shork.connect
+pub fn invalid_sql_test() {
+  let connection = start_default()
+
+  let sql = "insert update select *!!"
+
+  let assert Error(shork.ServerError(code, message)) =
+    shork.query(sql) |> shork.execute(connection)
 
-  let assert Error(res) =
-    shork.query("select * from no_table") |> shork.execute(connection)
+  should.equal(code, 1064)
 
-  should.equal(
-    shork.ServerError(1146, "Table 'poke.no_table' doesn't exist"),
-    res,
+  message
+  |> should.equal(
+    "You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near 'update select *!!' at line 1",
   )
 }
+
+pub fn insert_constraint_error_test() {
+  let connection = start_default()
+
+  let sql = shork.query("insert into shorks (id, name, size) values (?, ?, ?)")
+
+  let assert Error(shork.ServerError(code, message)) =
+    sql
+    |> shork.parameter(shork.int(1))
+    |> shork.parameter(shork.text("Hainelore"))
+    |> shork.parameter(shork.float(90.0))
+    |> shork.execute(connection)
+
+  should.equal(code, 1062)
+
+  message |> should.equal("Duplicate entry '1' for key 'shorks.PRIMARY'")
+}
+
+pub fn select_from_unknown_table_test() {
+  let connection = start_default()
+
+  let sql = "select * from unknown"
+
+  let assert Error(shork.ServerError(code, message)) =
+    shork.query(sql)
+    |> shork.execute(connection)
+
+  should.equal(code, 1146)
+
+  message |> should.equal("Table 'shork_test.unknown' doesn't exist")
+}
+
+pub fn insert_with_incorrect_types_test() {
+  let connection = start_default()
+  let sql =
+    "
+    insert into shorks (name, size) into (true, true)
+    "
+
+  let assert Error(shork.ServerError(code, message)) =
+    shork.query(sql) |> shork.execute(connection)
+
+  should.equal(code, 1064)
+
+  message
+  |> should.equal(
+    "You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near 'into (true, true)' at line 1",
+  )
+}
+
+pub fn transaction_commit_test() {
```

Comment:
Personally, I prefer smaller tests. But I'm not sure if that would be nitpicking. ^^"